### ISSUE TITLE
Race condition between dependency retrieval and filtered transaction receipt

### ISSUE THREAD

**Issue by devrandom** [bitcoinj/bitcoinj#1029]
Symptom: transaction stays in pending even though it was in a received filtered block.

Dependency retrieval in `Peer` is asynchronous.  While Peer is retrieving dependencies for a broadcast transaction, the download peer can send us the transaction hash in a filtered block.  But since the `Wallet` hasn't seen the transaction yet, it will ignore the filtered transaction hash.

**Comment by schildbach** [comment]
Related to https://github.com/bitcoinj/bitcoinj/issues/981?

**Comment by devrandom** [comment]
It is similar, but doesn't require the peer to fail in order to occur.  Also, I'm not sure I completely understand @mikehearn analysis there, will follow up.

**Comment by mikehearn** [comment]
Yes, makes sense. We should probably queue up processing of filtered blocks if a dependency download is currently in progress?
